support-frontend/app/controllers/Application.scala

```diff
@@ -422,6 +418,83 @@ class Application(
     }
   }
 
+  def getProductParamsFromContributionParams(
+      countryGroupId: String,
+      productCatalog: JsonObject,
+      queryString: Map[String, Seq[String]],
+  ) = {
+    val maybeSelectedContributionType = queryString
+      .get("selected-contribution-type")
+      .flatMap(_.headOption)
+      .map(_.toLowerCase)
+
+    val maybeSelectedAmount = queryString
+      .get("selected-amount")
+      .flatMap(_.headOption)
+      .map(s => Try(s.toDouble))
+      .flatMap(_.toOption)
+
+    val currency = countryGroupId match {
+      case "uk" => "GBP"
+      case "us" => "USD"
+      case "au" => "AUD"
+      case "eu" => "EUR"
+      case "int" => "USD"
+      case "nz" => "NZD"
+      case "ca" => "CAD"
+      case _ => "GBP"
+    }
+    val isAnnual = maybeSelectedContributionType.contains("annual")
+    val ratePlan = if (isAnnual) "Annual" else "Monthly"
+    val maybeSupporterPlusAmount = productCatalog.asJson.hcursor
```

Comment:
Do we not have a `ProductCatalog` case class we can deserialise this to?

Reply:
We've purposefully not defined this in Scala as we don't use it there - we just pass the JSON down from the API into the `window.guardian` object. 

We have the definitions for this API in TypeScript as that's how it's defined.

If we want Scala types - should we look at defining the API in a more generic way like Thrift to generate the models as maintaining a few sets of models if error prone and not great use of dev time?

Reply:
> Cursors are very low level to be working with in application code

@rupertbates not sure I get this - [from the docs](https://circe.github.io/circe/cursors.html) it looks like a general way to traverse JSON?

Reply:
It is general, that's kind of my point there's a whole load of unnecessary concepts in that code like cursor and downField and we're using strings which give us no type safety. I take your point about not needing it anywhere else currently though, lets leave it for now and we can improve it when we get to using the product catalog elsewhere in the stack.